scripts/build_ngx_pagespeed.sh

```diff
@@ -0,0 +1,331 @@
+    exit 1
+  fi
+
+  # If we set -e or -u then users of this script will see it silently exit on
+  # failure.  Instead we need to check the exit status of each command
+  # manually.  The run function (see below) handles exit-status checking for
+  # system-changing commands.
+
+  function usage() {
+    echo "
+Usage: build_ngx_pagespeed.sh [options]
+Options:
+  -v, --version <ngx_pagespeed version>
+      What version of ngx_pagespeed to build.  Required.
+
+  -n, --nginx <nginx version>
+      What version of nginx to build.  If not set, this script only prepares the
+      ngx_pagespeed module, and expects you to handle including it when you
+      build nginx.
+
+  -b, --builddir <directory>
+      Where to build.  Defaults to \$HOME.
+
+  -p, --depsinstalled
+      By default, this script checks for the packages it depends on and tries to
+      install them.  If you have installed dependencies from source or are on a
+      non-deb non-rpm system, this won't work.  In that case, you can install
+      the dependencies yourself and pass in --depsinstalled to skip this check.
+
+  -d, --dryrun
+      Don't make any changes to the system, just print what changes you
+      would have made.
+
+  -h, --help
+      Print this message and exit."
+  }
+
+  opts=$(getopt -o v:n:b:pdh \
+    --longoptions version:,nginx:,buildir:,depsinstalled,dryrun,help \
+    -n "$(basename "$0")" -- "$@")
+  if [ $? != 0 ]; then
+    usage
+    exit 1
+  fi
+  eval set -- "$opts"
+
+  NPS_VERSION=""
+  NGINX_VERSION=""
+  BUILDDIR="$HOME"
+  DEPSINSTALLED=false
+  DRYRUN=false
+  while true; do
+    case "$1" in
+      -v | --version) shift
+        NPS_VERSION="$1"
+        shift
+        ;;
+      -n | --nginx) shift
+        NGINX_VERSION="$1"
+        shift
+        ;;
+      -b | --builddir) shift
+        BUILDDIR="$1"
+        shift
+        ;;
+      -p | --depsinstalled) shift
+        DEPSINSTALLED=true
+        ;;
+      -d | --dryrun) shift
+        DRYRUN="true"
+        ;;
+      -h | --help) shift
+        usage
+        exit 0
+        ;;
+      --) shift
+        break
+        ;;
+      *)
+        echo "Invalid argument: $1"
+        usage
+        exit 1
+        ;;
+    esac
+  done
+
+  if [ -z "$NPS_VERSION" ]; then
+    echo "Please pass --version <ngx_pagespeed version>"
+    usage
+    exit 1
+  fi
+
+  # We pass all commands through run() so that in a dryrun we don't actually
+  # make any changes.
+  function run() {
+    if "$DRYRUN"; then
+      echo "would run $@"
+    else
+      if ! "$@"; then
+        echo "Failure running $@, exiting."
+        exit 1
+      fi
+    fi
+  }
+
+  function redhat_is_installed() {
+    local package_name="$1"
+    rpm -qa $package_name | grep -q .
+  }
+
+  function debian_is_installed() {
+    local package_name="$1"
+    dpkg -l $package_name | grep ^ii | grep -q .
+  }
+
+  function install_dependencies() {
+    local install="$1"
+    local is_installed="$2"
+    local dependencies="$3"
```

Comment:
I'd expect you to set install and is_install, then shift 2 and use "$@" for $dependencies.